liquibase-core/src/main/java/liquibase/changelog/StatusChangeLogIterator.java

```diff
@@ -0,0 +1,132 @@
+package liquibase.changelog;
+
+import liquibase.RuntimeEnvironment;
+import liquibase.Scope;
+import liquibase.change.core.TagDatabaseChange;
+import liquibase.changelog.filter.ChangeSetFilter;
+import liquibase.changelog.filter.ChangeSetFilterResult;
+import liquibase.changelog.filter.CountChangeSetFilter;
+import liquibase.changelog.filter.UpToTagChangeSetFilter;
+import liquibase.changelog.visitor.ChangeSetVisitor;
+import liquibase.changelog.visitor.SkippedChangeSetVisitor;
+import liquibase.exception.LiquibaseException;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class StatusChangeLogIterator extends ChangeLogIterator {
```

Comment:
Can you add some javadoc to this class and its methods?

Reply:
:+1: @wwillard7800 is gonna handle this one since this is from the DAT-13182 branch.